Public/Src/Engine/Processes/SandboxedProcessReports.cs

```diff
@@ -647,6 +648,44 @@ private bool FileAccessReportLineReceived<T>(ref T data, FileAccessReportProvide
                 }
             }
 
+            if (operation == ReportedFileOperation.FirstAllowWriteCheckInProcess)
+            {
+                // This operation represents that a given path was checked for write access for the first time
+                // within the scope of a process. The status of the operation represents whether that access should
+                // have been allowed/denied, based on the existence of the file.
+                // However, we need to determine whether to deny the access based on the first time the path was
+                // checked for writes across the whole process tree. This means checking the first time this operation 
+                // is reported for a given path, and ignore subsequent reports.
+                // Races are ignored: a race means two child processes are racing to create or delete the same file
+                // - something that is not a good build behavior anyway - and the outcome will be that we will 
+                // non -deterministically deny the access
+                if (path != null && !m_overrideAllowedWritePaths.ContainsKey(path))
+                {
+                    // We should override write allowed accesses for this path if the status of the special operation was 'denied'
+                    m_overrideAllowedWritePaths[path] = (status == FileAccessStatus.Denied);
+                }
+
+                return true;
+            }
+
+            FileAccessStatusMethod method;

```

Comment:
>FileAccessStatusMethod method; [](start = 12, length = 30)

`FileAccessStatusMethod method = FileAccessStatusMethod.PolicyBased;`

Thus you can reduce the unnecessary `else { ... }` branch.